proposed/dotnet-sdk-unified-config.md

```diff
@@ -0,0 +1,150 @@
+# A unified configuration file and experience for the .NET SDK
+
+**Owner** [Chet Husk](https://github.com/baronfel) | [Daniel Plaisted](https://github.com/dsplaisted)
+
+The .NET SDK is a product that consists of a number of different tools, all owned by different teams, all operating at slightly different levels of interaction. Some tools operate on the entire repository, some tools on a single project, and yet other tools on a single file.
+
+Because of the disparate nature of these tools, the configuration for each tool is different. This makes it difficult for users to understand how to configure the tools, and difficult for the tools to share configuration information. This document proposes a single, centralized document for repository-wide configuration that tools belonging to the SDK (as well as SDK-adjacent 1st- and 3rd-party tools) can use to share configuration information in a way that has
+
+* a consistent user experience
+* a well supported IDE experience
+* is shareable across team members and trackable in source control
+
+## What exists today
+
+Today there are several places where teams have put configuration for repo- or project-level configuration for tooling:
```

Comment:
May or may not be relevant, but I'd also consider `Directory.Build.rsp` as another of these types of configuration sources.

Reply:
Similarly, we'll need guidance for when to put something here versus `Directory.Build.props`. For instance, where should I put `TreatWarningsAsErrors`? I think that has a surprisingly big impact for MSBuild, given that one's a default property and one's a global property.